File: leetcode/Arrays/leetcode-26/26.go
package arrays

func removeDuplicates(nums []int) int {
	if nums == nil || len(nums) == 0 {
		return 0
	}

	p, t := 1, nums[0]

	for i := 1; i < len(nums); i++ {
		if nums[i] != t {
			nums[p] = nums[i]
			p++
			t = nums[i]
		}
	}
	return p
}

File: leetcode/Arrays/leetcode-26/26.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }

        int start = 1, tmp = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] != tmp) {
                nums[start++] = nums[i];
                tmp = nums[i];
            }
        }
        return start;
    }
}